actions/class.Creator.php

```diff
@@ -86,8 +89,24 @@ public function index()
                 )
             );
         }
+
         $this->setData('identifierUrl', _url('getIdentifier', null, null, ['uri' => $testUri]));
 
+        if ($this->isScaleEnabled()) {
+            $this->setData(
+                'scalesPresets',
+                json_encode(
+                    iterator_to_array(
+                        $this->getRemoteListService()->getListElements(
+                            $this->getClass(RemoteScaleListService::SCALES_URI)
+                        )
+                    )
+                )
+            );
```

Comment:
Ok this can fail on so many levels, can you split it to part where you build `$scalePresets` and handle all possible problems like empty list elements? Then do` json_encode` and also handle json errors?